v2/kafka-to-kafka/src/main/java/com/google/cloud/teleport/v2/options/KafkaToKafkaOptions.java

```diff
@@ -0,0 +1,180 @@
+      example = "topic1,topic2")
+  @Validation.Required
+  String getInputTopic();
+
+  void setInputTopic(String inputTopic);
+
+  @TemplateParameter.Text(
+      order = 3,
+      optional = false,
+      regexes = {"[,:a-zA-Z0-9._-]+"},
+      description = "output topics to write to",
+      helpText = "topics to write to for the data read from Kafka",
+      example = "topic1,topic2")
+  @Validation.Required
+  String getOutputTopic();
+
+  void setOutputTopic(String outputTopic);
+
+  @TemplateParameter.Text(
+      order = 4,
+      optional = false,
+      regexes = {"[,:a-zA-Z0-9._-]+"},
+      description = "sink kafka Bootstrap Server",
+      helpText = "sink kafka Bootstrap Server to write data to",
+      example = "localhost:9092")
+  @Validation.Required
+  String getSinkBootstrapServer();
+
+  void setSinkBootstrapServer(String sinkBootstrapServer);
+
+  @TemplateParameter.Enum(
+      order = 5,
+      enumOptions = {
+          @TemplateParameter.TemplateEnumOption("nonGMK-to-nonGMK"),
+          @TemplateParameter.TemplateEnumOption("GMK-to-GMK"),
+          @TemplateParameter.TemplateEnumOption("nonGMK-to-GMK")
+      },
+      optional = false,
+      description = "the type of kafka-to-kafka migration",
+      helpText = "Migration Type can be one of the three options `nonGMK-to-nonGMK`, `GMK-to-GMK`, nonGMK-to-GMK`"
+  )
+  String getMigrationType();
+  void setMigrationType(String migrationType);
+
+  @TemplateParameter.Text(
+      order = 6,
+      optional = true,
+      description = "secretid to get SASL username for kafka source",
+      helpText = "secretid for username to authenticate to kafka source"
+  )
+  String getSecretIdSourceUsername();
+  void setSecretIdSourceUsername(String secretIdSourceUsername);
+
+  @TemplateParameter.Text(
+      order = 7,
+      optional = true,
+      description = "secretid to get SASL password for the kafka source",
+      helpText = "secretid in secret manager to authenticate to kafka source"
+  )
+
+  String getSecretIdSourcePassword();
+  void setSecretIdSourcePassword(String secretIdSourcePassword);
+
+  @TemplateParameter.Text(
+      order = 8,
+      optional = true,
+      description = "secretid to get SASL username for kafka sink",
+      helpText = "secretid to get password to authenticate to kafka sink"
+  )
+  String getSecretIdSinkUsername();
+  void setSecretIdSinkUsername(String secretIdSinkUsername);
+  @TemplateParameter.Text(
+      order = 9,
+      optional = true,
+      description = "secretid to get password for the kafka sink",
+      helpText = "secretid in secret manager to get access to the secret"
+  )
+  String getSecretIdSinkPassword();
+  void setSecretIdSinkPassword(String secretIdSinkPassword);
+
+  @TemplateParameter.Text(
+      order = 10,
+      optional = true,
+      description = "versionid from the secret manager to get username for source",
+      helpText = "version of the secret"
+  )
+
+  String getVersionIdSourceUsername();
```

Comment:
As discussed offline, for Username and other params, we should only have a single "secretId" parameter.